src/lib/caching/structures/channels/NewsChannel.ts

```diff
@@ -16,6 +16,21 @@ export class NewsChannel extends GuildTextChannel {
 	 */
 	public readonly type = ChannelType.GuildAnnouncement;
 
+	/**
+	 * Follows this channel.
+	 * @param webhookChannel
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/channel#channel-object-channel-types
+	 */
+	public async follow(webhookChannel: GuildTextChannel): Promise<APIChannelFollowResult> {
+		const reqData = {
+			data: {
+				webhook_channel_id: webhookChannel.id
+			}
+		};
+		return this.client.api.post(Routes.followChannel(this.id), reqData) as unknown as APIChannelFollowResult;
```

Comment:
I don't think you need the `as unknown` part

Reply:
I supposedly do as tsc throws this error at me

```
Conversion of type 'Promise<unknown>' to type 'APIChannelFollowResult' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'Promise<unknown>' is missing the following properties from type 'APIChannelFollowResult': channel_id, webhook_id
```

Reply:
Promise<APIChannelFollowResult>